server/src/main/java/com/cloud/tags/TaggedResourceManagerImpl.java

```diff
@@ -318,7 +318,11 @@ public Map<String, String> getTagsFromResource(ResourceObjectType type, long res
     private void informStoragePoolForVmTags(long vmId, String key, String value) {
         List<VolumeVO> volumeVos = volumeDao.findByInstance(vmId);
         for (VolumeVO volume : volumeVos) {
-            DataStore dataStore = dataStoreMgr.getDataStore(volume.getPoolId(), DataStoreRole.Primary);
+            Long poolId = volume.getPoolId();
+            if (poolId == null) {
+                continue;
+            }
+            DataStore dataStore = dataStoreMgr.getDataStore(poolId, DataStoreRole.Primary);
             if (dataStore == null || !(dataStore.getDriver() instanceof PrimaryDataStoreDriver)) {
                 continue;
             }
```

Comment:
can you combine these and put them in a separate method?
something like 
```
DataStore dataStore = null;
try {
    DataStore dataStore = retrieveDatastore(poolId);
} catch (<local>Exception e) {
   log.info(String.format("no data store found for vm %s with poolId %d", vm.getName(), poolId));
    continue;
}
```
and move the code to new method `DataStore retrieveDatastore(Long poolId) {..}`

Reply:
Hi Daan, I extracted it. I'm not sure I follow what the goal is with the throwing. It would still require a null check after the catch and another `continue`.

Reply:
yeah, it was just an example. Your solution is addequate.